build_from.py

```diff
@@ -130,6 +130,41 @@ def build_by_notice(filename, title, act_title, act_section,
     if args.generate_diffs:
         generate_diffs(reg_tree, act_title_and_section, builder, layer_cache)
 
+
+def generate_xml(filename, title, act_title, act_section, notice_doc_numbers,
+                 doc_number=None, checkpoint=None):
+
+    act_title_and_section = [act_title, act_section]
+    #   First, the regulation tree
+
+    print filename, title, act_title_and_section
+
+    reg_tree, builder = tree_and_builder(filename, title,
+                                         checkpoint, writer_type='XML')
+    layer_cache = LayerCacheAggregator()
+    layers = builder.generate_layers(reg_tree, act_title_and_section, layer_cache)
+
+    # builder.write_notices()
+
+    #   Always do at least the first reg
+    logger.info("Version %s", builder.doc_number)
+    builder.write_regulation(reg_tree, output_type='xml', layers=layers)
+    all_versions = {doc_number: FrozenNode.from_node(reg_tree)}
+
+    for last_notice, old, new_tree, notices in builder.revision_generator(
+            reg_tree):
```

Comment:
Another note-to-selves, when we're refactoring the build process as a whole, I'd want to be able to run this on just a single notice (say, a new notice for Z), and load the old tree, notices, etc, from XML. Again, future work for now. I think the decision to mirror the current process as much as possible is a good one that keeps us from debugging too many things all at once.